python-main.js

```diff
@@ -1080,37 +1087,58 @@ function web_editor(config) {
             $("#command-flash > .roundlabel").text(config["translate"]["static-strings"]["buttons"]["command-flash"]["label"]);
             $("#command-flash").attr("title", config["translate"]["static-strings"]["buttons"]["command-flash"]["title"]);
 
-            if (serial){
-              doSerial();
+            if (serial) {
+                doSerial();
             }
         })
-        .catch(function(err) {
-            console.log("Error connecting: " + err);
-            $("#flashing-overlay-container").css("display", "flex");
-            $("#flashing-info").addClass('hidden');
-
-            // If micro:bit does not support dapjs
-            if (err.message === "No valid interfaces found."){
-                $("#flashing-overlay-error").html('<div>' + err + '</div><div>'+ config["translate"]["webusb"]["err"]["update-req"] +'</div><a href="#" onclick="flashErrorClose()">'+ config["translate"]["webusb"]["close"] +'</a>');
-                return;
-            } else if (err.message === "Unable to claim interface.") {
-                $("#flashing-overlay-error").html('<div>'+ config["translate"]["webusb"]["err"]["clear-connect"] +'</div><a href="#" onclick="flashErrorClose()">'+ config["translate"]["webusb"]["close"] +'</a>');
-                return;
-            }
+        .catch(webusbErrorHandler);
+    }
 
-            $("#flashing-overlay-error").html('<div>' + err + '</div><div>'+ config["translate"]["webusb"]["err"]["restart-microbit"] +'</div><a href="#" onclick="flashErrorClose()">'+ config["translate"]["webusb"]["close"] +'</a>');
-        });
+    function webusbErrorHandler(err) {
+        // Disconnect
+        doDisconnect();
+
+        // Error handler
+        $("#flashing-overlay-container").css("display", "flex");
+        $("#flashing-info").addClass('hidden');
+
+        // If micro:bit does not support dapjs
+        var errorMessage;
+        if (err.message === "No valid interfaces found."){
+            errorMessage = "update-req";
+        } else if (err.message === "Unable to claim interface.") {
+            errorMessage = "clear-connect";
+        } else {
```

Comment:
We should also add a check for "device disconnected" here to customise the message shown to the user.

Then the micro:bit is physically unplugged while it is connected, this is what it shows:
![image](https://user-images.githubusercontent.com/29712657/63355565-3bda1180-c35e-11e9-94df-e3fc0bbc03c6.png)

The first line is good, but the second is not quite appropriate as the user was probably not doing anything in the editor to "try again".